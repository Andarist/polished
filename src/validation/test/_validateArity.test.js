// @flow
import validateArity from '../_validateArity'

describe('validateArity', () => {
  const modulePath = 'module/testModule'

  beforeAll(() => {
    global.console = {
      error: jest.fn(),
      warn: jest.fn(),
      log: global.console.log,
    }
  })

  afterEach(() => {
    // eslint-disable-next-line no-console
    console.error.mockClear()
    // eslint-disable-next-line no-console
    console.warn.mockClear()
  })

  // Exactly
  it('should return true when passed exactly the arguments needed', () => {
    expect(
      validateArity(
        modulePath,
        {
          type: 'string',
        },
        [1],
      ),
    ).toBeTruthy()
  })

  it('should return true when passed more arguments than needed', () => {
    expect(
      validateArity(
        modulePath,
        {
          type: 'string',
        },
        [1, 2],
      ),
    ).toBeTruthy()
  })

  it('should throw a warning when passed more arguments than needed', () => {
    validateArity(
      modulePath,
      {
        type: 'string',
      },
      [1, 2],
    )
    // eslint-disable-next-line no-console
    expect(console.warn.mock.calls).toMatchSnapshot()
    // eslint-disable-next-line no-console
    console.warn.mockClear()
  })

  it('should return true when passed less arguments than needed', () => {
    expect(
      validateArity(modulePath, [{ type: 'string' }, { type: 'string' }], [1]),
    ).toBeTruthy()
  })

  it('should return true when not passed any types', () => {
    expect(validateArity(modulePath, undefined, [1])).toBeTruthy()
  })

  it('should throw a properly formatted warning when difference in expected and actual is not equal to 1', () => {
    validateArity(
      modulePath,
      {
        type: 'string',
      },
      [1, 2, 3, 4],
    )
    // eslint-disable-next-line no-console
    expect(console.warn.mock.calls).toMatchSnapshot()
    // eslint-disable-next-line no-console
    console.warn.mockClear()
  })
})
