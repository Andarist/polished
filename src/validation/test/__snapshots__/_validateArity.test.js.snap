exports[`validateArity should throw a properly formatted warning when difference in expected and actual is not equal to 1 1`] = `
Array [
  Array [
    "%c ✨ WARNING ✨ ---- module/testModule.js --%c

testModule %cexpects a maximum of %c1 parameter%c. However, you passed %c4 parameters%c. These additional parameters were ignored.
  %c
Please see the documentation at %chttps://polished.js.org/docs/#testModule%c for more information.",
    "font-weight: bold; color: black",
    "color: black; font-size: 12px; font-weight: bold",
    "color: black; font-size: 12px",
    "color: black; font-size: 12px; font-weight: bold; color: green",
    "color: black; font-size: 12px",
    "color: black; font-size: 12px; font-weight: bold; color: goldenrod",
    "color: black; font-size: 12px",
    "color: gray; line-height: 1.4",
    "color: blue; line-height: 1.4",
    "color: gray; line-height: 1.4",
  ],
]
`;

exports[`validateArity should throw a warning when passed more arguments than needed 1`] = `
Array [
  Array [
    "%c ✨ WARNING ✨ ---- module/testModule.js --%c

testModule %cexpects a maximum of %c1 parameter%c. However, you passed %c2 parameters%c. This additional parameter was ignored.
  %c
Please see the documentation at %chttps://polished.js.org/docs/#testModule%c for more information.",
    "font-weight: bold; color: black",
    "color: black; font-size: 12px; font-weight: bold",
    "color: black; font-size: 12px",
    "color: black; font-size: 12px; font-weight: bold; color: green",
    "color: black; font-size: 12px",
    "color: black; font-size: 12px; font-weight: bold; color: goldenrod",
    "color: black; font-size: 12px",
    "color: gray; line-height: 1.4",
    "color: blue; line-height: 1.4",
    "color: gray; line-height: 1.4",
  ],
]
`;

exports[`validateArrity should throw a properly formatted warning when difference in expected and actual is not equal to 1 1`] = `
Array [
  Array [
    "%c ✨ WARNING ✨ ---- module/testModule.js --%c

testModule %cexpects a maximum of %c1 parameter%c. However, you passed %c4 parameters%c. These additional parameters were ignored.
  %c
Please see the documentation at %chttps://polished.js.org/docs/#testModule%c for more information.",
    "font-weight: bold; color: black",
    "color: black; font-size: 12px; font-weight: bold",
    "color: black; font-size: 12px",
    "color: black; font-size: 12px; font-weight: bold; color: green",
    "color: black; font-size: 12px",
    "color: black; font-size: 12px; font-weight: bold; color: goldenrod",
    "color: black; font-size: 12px",
    "color: gray; line-height: 1.4",
    "color: blue; line-height: 1.4",
    "color: gray; line-height: 1.4",
  ],
]
`;

exports[`validateArrity should throw a warning when passed more arguments than needed 1`] = `
Array [
  Array [
    "%c ✨ WARNING ✨ ---- module/testModule.js --%c

testModule %cexpects a maximum of %c1 parameter%c. However, you passed %c2 parameters%c. This additional parameter was ignored.
  %c
Please see the documentation at %chttps://polished.js.org/docs/#testModule%c for more information.",
    "font-weight: bold; color: black",
    "color: black; font-size: 12px; font-weight: bold",
    "color: black; font-size: 12px",
    "color: black; font-size: 12px; font-weight: bold; color: green",
    "color: black; font-size: 12px",
    "color: black; font-size: 12px; font-weight: bold; color: goldenrod",
    "color: black; font-size: 12px",
    "color: gray; line-height: 1.4",
    "color: blue; line-height: 1.4",
    "color: gray; line-height: 1.4",
  ],
]
`;
